src/documentation/system_administrators/advanced/quantum_safe_storage/quantum_safe_storage_deploy.md

```diff
@@ -0,0 +1,594 @@
+
+Open a file `zstor-default.toml` in a text editor and paste in the template below. For example:
+
+```sh
+$EDITOR zstor-default.toml
+```
+
+Initial contents of the file:
+
+```
+minimal_shards = 16
+expected_shards = 20
+redundant_groups = 0
+redundant_nodes = 0
+zdbfs_mountpoint = "/mnt/qsfs"
+socket = "/var/run/zstor.sock"
+prometheus_port = 9100
+zdb_data_dir_path = "/data/data/zdbfs-data"
+max_zdb_data_dir_size = 25600
+
+[encryption]
+algorithm = "AES"
+key = "Write your key here"
+
+[compression]
+algorithm = "snappy"
+
+[meta]
+type = "zdb"
+
+[meta.config]
+prefix = "qsfs-meta"
+
+[meta.config.encryption]
+algorithm = "AES"
+key = "write your key here"
+```
+
+Make sure to edit the file as needed. You can change the minimal and expected shards according to your own plan. Another important value is `max_zdb_data_dir_size`, which is how large the cache is allowed to grow before data blocks are removed. This value is given in MiB. Therefore, the example shown is 25GiB.
+
+It is also necessary to fill in the encryption keys with your own. You can use the same or different keys for data and metadata, at your own preference. The key must be 32 bytes in hex format. Here's an example of how to generate a key in this format:
+
+```sh
+openssl rand -hex 32
+```
+
+Once this is done, take the output and insert it into the file in the indicated locations.
+
+### Prepare for Zdb Deployment
+
+Now we will deploy the zdbs and write their details into the config file in the proper format. Before proceeding, we'll set a few shell variables that will be used in the deployment scripts:
+
+```sh
+CONFIG=zstor-default.toml
+PASSWORD=$(openssl rand -base64 18)
+METADATA_NODES="1 2 3 4"
+back end_NODES="1 2 3 4 5 6 7 8"
+back end_SIZE=1
+```
+
+This will generate a strong random password that will be used to secure each zdb. You can replace the code that generates the password with your own password if you wish. For now, don't worry about having to save the randomly generated password. It will get written to the config file and you can take note of it later.
+
+For the metadata and back end nodes, replace the example values with the node IDs you selected before. Set your desired back end size too, which is specified in gigabtyes.
+
+
+### Deploy Metadata Zdbs
+
+Here is an example script to deploy the metadata zdbs based on the variables set above. This uses a fixed size of 1 GB, which should be plenty:
+
+```sh
+for node in $METADATA_NODES; do
+  name=node${node}meta
+  tfcmd deploy zdb --mode user --node $node -n $name --size 1 --password $PASSWORD
+done
+```
+
+It's possible that some of your chosen nodes don't respond properly at deployment time and need to be replaced with other nodes. In that case, just replace the variable with the new node IDs like this:
+
+```sh
+# Example: Node 3 wasn't working, replace it with node 5
+METADATA_NODES="1 2 4 5"
```

Comment:
Why 4 nodes? Can't it be less or more?

Reply:
This is covered in the previous section on planning. I'll mention that it's there for reference though.